packages/AdaptiveCards/Actions/SendAdaptiveCard.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder.Dialogs;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Components.AdaptiveCards
+{
+    public class SendAdaptiveCard : BaseAdaptiveCard
+    {
+
+        [JsonProperty("$kind")]
+        public const string Kind = "Microsoft.Bot.Components.SendAdaptiveCard";
+
+        [JsonProperty("activityIdProperty")]
+        public StringExpression ActivityIdProperty { get; set; }
+
+        protected async override Task<object> OnProcessCardAsync(DialogContext dc, JObject card, CancellationToken cancellationToken = default)
+        {
+            // Send activity to client
+            var activity = CreateActivity(dc, card);
+            var response = await dc.Context.SendActivityAsync(activity, cancellationToken).ConfigureAwait(false);
+
+            // Get activity ID (if there is one)
+            var activityId = response != null && !String.IsNullOrEmpty(response.Id) ? response.Id : String.Empty;
```

Comment:
Shouldn't there always be an ID in the sent activity? I thought it was required per the protocol?

Reply:
No... You can get null back in several situations. Skills running in "expectReplies" mode for instance. You definitely shouldn't assume you got back an ID.

---
In reply to: [610219671](https://github.com/microsoft/botframework-components/pull/862#discussion_r610219671) [](ancestors = 610219671)